core/src/main/java/io/grpc/internal/ServiceConfigUtil.java

```diff
@@ -178,7 +183,6 @@ static Double getBackoffMultiplierFromRetryPolicy(Map<String, ?> retryPolicy) {
     String retryableStatusCodesKey = "retryableStatusCodes";
     Set<Status.Code> codes = getListOfStatusCodesAsSet(retryPolicy, retryableStatusCodesKey);
     verify(codes != null, "%s is required in retry policy", retryableStatusCodesKey);
```

Comment:
So it must be provided, but just make it empty? I thought it was now completely optional.

I realize now that removing these two constraints does cause deployment issues as if an old client receives a new config then the config will be considered invalid. Unclear how to address that.

Reply:
That's a problem. But retry was experimental and only if retry is enabled we parse the retry config https://github.com/grpc/grpc-java/blob/v1.39.0/core/src/main/java/io/grpc/internal/ManagedChannelServiceConfig.java#L283

So only old clients trying the experimental feature, they will have validation error for new config. Old client not enabling retry will be fine.

Reply:
Ah, good point. Dunno if that applies to Go as well.